src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -85,6 +95,8 @@ public void AllowStop()
 
         public void Stop(TimeSpan timeout)
         {
+            DisposeConnections(timeout);
```

Comment:
This should probably be after `_thread.IsAlive`. I would expect both the `WalkConnectionsAndClose` and `PostAsync` in `DisposeConnections` to hang if the thread is dead.